src/Illuminate/Console/Concerns/HasParameters.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Illuminate\Console\Concerns;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+
+trait HasParameters
+{
+    /**
+     * Specify the arguments and options on the command.
+     *
+     * @return void
+     */
+    protected function specifyParameters()
+    {
+        // We will loop through all of the arguments and options for the command and
+        // set them all on the base command instance. This specifies what can get
+        // passed into these commands as "parameters" to control the execution.
+        foreach ($this->getArguments() as $arguments) {
+            if ($arguments instanceof InputArgument) {
+                $this->getDefinition()->addArgument($arguments);
+            } else {
+                call_user_func_array([$this, 'addArgument'], $arguments);
```

Comment:
The trait needs to have `getDefinition`, `addArgument` and `addOption` as abstract methods.

Reply:
`addArgument` and `addOption` both will have scalar type-hinted in 5.0. Does Laravel want to have this?

Reply:
Laravel 6 requires Symfony console 4, so no typehints needed. For Laravel 7, yes, we'll need to match the symfony method signature, since that needs Symfony console 5.

Reply:
The same as some of Laravel's internals has typehints to match phpunit's stuff.